src/tests/testCacheManager.cc

```diff
@@ -123,7 +110,9 @@ TestCacheManager::testRegister()
     CPPUNIT_ASSERT_EQUAL(false, profile->isAtomic);
     CPPUNIT_ASSERT_EQUAL(String("sample"), String(action->name()));
 
-    StoreEntry *sentry=new StoreEntry();
+    StoreEntry *sentry = new StoreEntry();
+    sentry->createMemObject();
+    sentry->mem_obj->setUris("http://localhost/squid-internal-mgr/menu", "http://localhost/squid-internal-mgr/menu", HttpRequestMethod(Http::METHOD_GET));
     sentry->flags=0x25; //arbitrary test value
```

Comment:
BTW, this terrible flags hack should be removed -- Store relies on proper entry flags! If needed, dummy_action() can set a global static "dummyActionRan" variable that testRegister() can check. I am not saying that this PR should fix this test, but if this hack is in the way of this PR, than this PR may fix this test.

Reply:
I don't think it is in the way of this PR